Add support for LinearOpContext and Conv2dOpContext in quantization pass

jiashenC opened the pull request:
Summary:
Expose unpacking method for Linear/Conv2dOpContext and support linear_clamp_run and conv2d_clamp_run.

```
| suite   |   #models |   #has_ts_model |   #has_sample_inputs |   #ts_can_run |   #can_convert |   #ep_result_correct |   #aoti_can_compile |   #can_package |   #sigmoid_can_run |   #sigmoid_result_correct |
|---------|-----------|-----------------|----------------------|---------------|----------------|----------------------|---------------------|----------------|--------------------|---------------------------|
| EDGEML  |        50 |              36 |                   32 |            30 |             19 |                    7 |                   0 |             19 |                 16 |                         6 |
```

Test Plan:
```
buck2 run mode/dev-nosan sigmoid/inference/ts_migration:main -- --mode test_all --test_suites edgeml
```

Reviewed By: angelayi

Differential Revision: D60871242

ydwu4 commented on aten/src/ATen/native/xnnpack/RegisterOpContextClass.cpp:
```diff
@@ -62,9 +64,9 @@ TORCH_LIBRARY(xnnpack, m) {
               std::move(std::get<3>(state)),
```
Not familiar with this. Why we remove std::move for these? Can add a comment?

jiashenC replied:
This is actually suggested by the internal lint. Something like `std::move` does not have any effect when we are returning a `const`.

ydwu4 replied:
ok, can we remove all of the "move"? Seems we didn't treat all the fields uniformly.

ydwu4 reviewed (commented):
Add a unit test?

Commit:
Add support for LinearOpContext and Conv2dOpContext in quantization pass (#133098)

Summary:
Expose unpacking method for Linear/Conv2dOpContext and support linear_clamp_run and conv2d_clamp_run.

```
| suite   |   #models |   #has_ts_model |   #has_sample_inputs |   #ts_can_run |   #can_convert |   #ep_result_correct |   #aoti_can_compile |   #can_package |   #sigmoid_can_run |   #sigmoid_result_correct |
|---------|-----------|-----------------|----------------------|---------------|----------------|----------------------|---------------------|----------------|--------------------|---------------------------|
| EDGEML  |        50 |              36 |                   32 |            30 |             19 |                    7 |                   0 |             19 |                 16 |                         6 |
```

Pull Request resolved: https://github.com/pytorch/pytorch/pull/133098

Test Plan:
```
buck2 run mode/dev-nosan sigmoid/inference/ts_migration:main -- --mode test_all --test_suites edgeml
```

Reviewed By: angelayi

Differential Revision: D60871242

Pulled By: jiashenC

angelayi commented:
landed with https://github.com/pytorch/pytorch/pull/133622
